### ISSUE TITLE
Cannot merge backported PRs because "bedevere/maintenance-branch-pr" not run

### ISSUE THREAD

**Issue by ned-deily** [python/core-workflow#370]
See, for example, python/cpython#20305 and python/cpython#20306.  For some reason, it seem that the bedevere/maintenance-branch-pr check did not run (yet); the status is: Expected -Waiting for status to be reported.  Because it is a required check, the PRs can not be merged with manually or automatically.

**Comment by encukou** [comment]
I noticed the bot [reacts to edits](https://github.com/python/bedevere/blob/ba44dbc36e37c2f4ba1436bfc345533837a49569/bedevere/backport.py#L72), so I edited the titles of those two PRs slightly to get them merged.
I don't know why the bot didn't wake up at first :(

**Comment by ned-deily** [comment]
Oh, thanks, @encukou ! Still, somebody should look at this. I don't recall ever seeing this behavior before.

**Comment by Mariatta** [comment]
I could see several few timeout errors from the webhook deliveries, so the bot could not perform the task. For those with administrative power, they can go to Cpython's repo > Settings tab > Webhoks and click on the webhook for bedevere. Normally, redelivering the webhook would fix it.

Honestly I don't know what we can do from our side to fix the problem. But I have seen this happening a few times.


<img width="770" alt="Screen Shot 2020-05-21 at 4 48 59 PM" src="https://user-images.githubusercontent.com/5844587/82616794-11361700-9b83-11ea-9308-de487bef73c5.png">

**Comment by brettcannon** [comment]
The vast majority of bot issues are due to a failure to deliver the webhook event, and that usually lies somewhere between GitHub and Heroku. As such, typically the easiest thing to do it do something which triggers another. event (e.g. edits, opening and closing, etc.). Otherwise you can also check the commit history of the bots to see if something recently changed (which is usually not the case).

**Comment by Mariatta** [comment]
Re: timeouts in Heroku, I see now that some requests to bedevere are taking more than 30ms which caused the timeout. But I still don't actually know what to do to avoid the timeout.

I wonder if it's a matter of upgrading the dyno?

**Comment by Mariatta** [comment]
screenshot from heroku  dashboard
![Screen Shot 2020-05-25 at 10 44 47 AM](https://user-images.githubusercontent.com/5844587/82834202-44bfbc80-9e75-11ea-80b8-c39995a608f8.png)

**Comment by brettcannon** [comment]
I wonder if it's from a dyno spinning up? Do we have permanently running dynos?

**Comment by Mariatta** [comment]
We are using the hobby dyno. Perhaps we need one of the Professional Dynos?
https://www.heroku.com/pricing

We can probably take a closer look at webhook events that have been timing out and see how to we can improve the performance.

**Comment by smitthakkar96** [comment]
If it's the matter of hosting a webhook have you considered using something
like https://workers.cloudflare.com/ or AWS Lambda, I bet it would be much
cheaper or might fit the free tier. Let me know if I could help with it, I
would be happy to help.

Thanks,
Smit

**Comment by Mariatta** [comment]
The PSF receive Heroku credits and most of our webservices are on Heroku whenever possible. Perhaps PSF infrastructure team can weigh in cc @ewdurbin

**Comment by ewdurbin** [comment]
Yes, we utilize donated services (~$750 service credit per month) from Heroku for all of our bots at the moment. Hobby dynos should not sleep.

If we think its an issue with resources and not some other factor (code, db, or blocking I/O) we can upgrade the dynos... but ultimately that gets very expensive ($7/mo in credits to $25/mo in credits). If we think it would help... it might require consolidating our fleet of bots into a single app instance somehow.

**Comment by smitthakkar96** [comment]
Shouldn’t we consider contacting AWS or GCP guys and see if we can get some
free credits from them, I am sure they would be glad to support PSF.  I
believe running the webhooks on serverless service like lambda or cloud
functions will be considerably cheaper and we might not require to
consolidate bots into a single app instance. Thoughts?

**Comment by ewdurbin** [comment]
We do also have service credits with other cloud vendors. If there is a volunteer that is willing to port and maintain the bots on a serverless platform, that could work.

At the moment, Heroku provides a common platform and easy ACL for contributors. PSF Infrastructure can support these additional needs if the current maintainers are on board.

**Comment by brettcannon** [comment]
I'm personally not interested in migrating to another platform unless it's Azure (since I already know it and can get help internally), or moving the check to GitHub Actions (since it's even more accessible to people and I'm also familiar with it).

**Comment by smitthakkar96** [comment]
Moving check to Github actions sounds like an amazing idea. If there is
something that Github actions cannot do I guess we can consider using
serverless offerings of Microsoft Azure if that’s something the team wants.
I can help with it (any porting required plus setup on Azure), we can use
Terraform or Pulumi to provision the resources on Azure. There is support
for IAM on Azure plus there are lot of open source projects that can help
integrate Github teams with Azure IAM.

Thanks,
Smit
